src/lib/provable/crypto/foreign-curve.ts

```diff
@@ -18,7 +18,7 @@ import { Bytes } from '../bytes.js';
 export { createForeignCurve, ForeignCurve };
 
 // internal API
-export { toPoint, FlexiblePoint };
+export { toPoint, FlexiblePoint, ForeignCurveNotNeeded };
```

Comment:
By exporting `ForeignCurveNotNeeded`, we make it linkable in other comments, where it is already in use.  This makes the whole chain public.

`ForeignCurveNotNeeded` in particular seems weird.  I'm not sure if we should just delete it in favor of `ForeignCurve`.  See: [this PR](https://github.com/o1-labs/o1js/pull/1757/files#diff-e2b15bfc1f6103679925c6ccf6697f5877d73e219b0433976dfba6111cf45febR280) which introduces it.